internal/pkg/agent/vault/vault_notdarwin.go

```diff
@@ -0,0 +1,214 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build !darwin
+
+package vault
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/gofrs/flock"
+)
+
+const (
+	// defaultFlockRetryDelay default file lock retry delay
+	defaultFlockRetryDelay = 10 * time.Millisecond
+
+	// lock file name
+	lockFile = `.lock`
+)
+
+type Vault struct {
+	path string
+	seed []byte
+
+	lockRetryDelay time.Duration
+	lock           *flock.Flock
+}
+
+// New creates the vault store
+func New(ctx context.Context, path string, opts ...OptionFunc) (v *Vault, err error) {
+	options := applyOptions(opts...)
+	dir := filepath.Dir(path)
+
+	// If there is no specific path then get the executable directory
+	if dir == "." {
+		exefp, err := os.Executable()
+		if err != nil {
+			return nil, fmt.Errorf("could not get executable path: %w", err)
+		}
+		dir = filepath.Dir(exefp)
+		path = filepath.Join(dir, path)
+	}
+
+	if options.readonly {
+		fi, err := os.Stat(path)
+		if err != nil {
+			return nil, err
+		}
+		if !fi.IsDir() {
+			return nil, fs.ErrNotExist
+		}
+	} else {
+		err := os.MkdirAll(path, 0750)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create vault path: %v, err: %w", path, err)
+		}
+		err = tightenPermissions(path)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	r := &Vault{
+		path:           path,
+		lockRetryDelay: options.lockRetryDelay,
+		lock:           flock.New(filepath.Join(path, lockFile)),
+	}
+
+	err = r.tryLock(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err = r.unlock(err)
```

Comment:
[Blocker]
It'll override the error of `getOrCreateSeed`. Better to `errors.Join` the errors.

Reply:
it will not, it's a helper func using errors.Join